scipy/stats/stats.py

```diff
@@ -2829,7 +2829,12 @@ def median_absolute_deviation(x, axis=0, center=np.median, scale=1.4826,
 
     # Consistent with `np.var` and `np.std`.
     if not x.size:
-        return np.nan
+        nan_shape = [item for i, item in enumerate(x.shape) if i != axis]
+        nan_array = np.full(nan_shape, np.nan)
+        if not nan_array.size:
```

Comment:
This doesn't quite work.  If there is a 0 in `nan_array.shape`, `nan_array.size` will be 0, and this will return the scalar `np.nan`, even if `nan_shape` is a nontrivial tuple.  For example, `median_absolute_deviation(np.zeros((3, 0, 5)), axis=-1)` should return an array with shape (3, 0).

Reply:
I will try to fix it.

Reply:
PTAL #11747